src/loader/bgw_counter.c

```diff
@@ -111,11 +110,44 @@ bgw_total_workers_increment()
 extern void
 bgw_total_workers_decrement()
 {
+	/*
+	 * Launcher is 1 worker, and when it dies we reinitialize, so we should
+	 * never be below 1
+	 */
+	SpinLockAcquire(&ct->mutex);
+	if (ct->total_workers > 1)
+		ct->total_workers--;
+	else
+		ereport(ERROR, (errmsg("cannot decrement workers below 1")));
+	SpinLockRelease(&ct->mutex);
+}
+
+extern bool
+bgw_total_workers_multi_increment(int increment_by)
+{
+	bool		incremented = false;
+	int			max_workers = guc_max_background_workers;
+
 	SpinLockAcquire(&ct->mutex);
-	ct->total_workers--;
+	if (ct->total_workers + increment_by <= max_workers)
+	{
+		ct->total_workers += increment_by;
+		incremented = true;
+	}
 	SpinLockRelease(&ct->mutex);
+	return incremented;
 }
 
+extern void
+bgw_total_workers_multi_decrement(int decrement_by)
+{
+	SpinLockAcquire(&ct->mutex);
+	if (ct->total_workers - decrement_by >= 1)
+		ct->total_workers -= decrement_by;
+	else
+		ereport(ERROR, (errmsg("cannot decrement workers below 1")));
```

Comment:
comment/explanation from `bgw_total_workers_decrement` should be duplicated here